src/main/java/gg/solarmc/kitpvp/kill/DamageHolder.java

```diff
@@ -24,26 +19,30 @@ public DamageHolder(KitpvpPlugin plugin) {
      * Adds a player to the damagers and then removes them 10 seconds later. This is probably shitcode in someone's book but tbh at this point i don't care.
      * @param damager the one who damages
      */
-    public void damage(Player damager) {
+    public void damage(UUID damager) {
```

Comment:
Just so you know, it's fine to use either Player or UUID for method and constructor parameters. Player grants you slightly more flexibility too.

What matters is that the *implementation* does not persist Player instances, that's all.